pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,125 @@
+package monitoringstack
+
+import (
+	"fmt"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/rhobs/observability-operator/pkg/apis/monitoring/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	AvailableReason                = "MonitoringStackAvailable"
+	ReconciledReason               = "MonitoringStackReconciled"
+	PrometheusNotAvailable         = "PrometheusNotAvailable"
+	PrometheusNotReconciled        = "PrometheusNotReconciled"
+	CannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
+	AvailableMessage               = "Monitoring Stack is available"
+	ReconciledMessage              = "Monitoring Stack successfully reconciled"
+)
+
+func updateConditions(msConditions []v1alpha1.Condition, promConditions []monv1.PrometheusCondition, generation int64, recError error) []v1alpha1.Condition {
+	if len(msConditions) == 0 {
+		return []v1alpha1.Condition{
+			{
+				Type:               v1alpha1.AvailableCondition,
+				Status:             v1alpha1.ConditionUnknown,
+				Reason:             "None",
+				LastTransitionTime: metav1.Now(),
+			},
+			{
+				Type:               v1alpha1.ReconciledCondition,
+				Status:             v1alpha1.ConditionUnknown,
+				Reason:             "None",
+				LastTransitionTime: metav1.Now(),
+			},
+		}
+	}
+	var updatedConditions []v1alpha1.Condition
+
+	for _, mc := range msConditions {
+		switch mc.Type {
+		case v1alpha1.AvailableCondition:
+			available := updateAvailable(mc, promConditions, generation)
+			if !available.Equal(mc) {
+				available.LastTransitionTime = metav1.Now()
+			}
+			updatedConditions = append(updatedConditions, available)
+		case v1alpha1.ReconciledCondition:
+			reconciled := updateReconciled(mc, promConditions, generation, recError)
+			if !reconciled.Equal(mc) {
+				reconciled.LastTransitionTime = metav1.Now()
+			}
+			updatedConditions = append(updatedConditions, reconciled)
+		}
+	}
+
+	return updatedConditions
+}
+
+// updateAvailable gets existing "Available" condition and updates its parameters
+// based on the Prometheus "Available" condition
+func updateAvailable(ac v1alpha1.Condition, prometheusConditions []monv1.PrometheusCondition, generation int64) v1alpha1.Condition {
+	ac.ObservedGeneration = generation
+
+	prometheusAvailable, err := getPrometheusCondition(prometheusConditions, monv1.PrometheusAvailable)
+
+	if err != nil {
+		ac.Status = v1alpha1.ConditionUnknown
+		ac.Reason = PrometheusNotAvailable
+		ac.Message = CannotReadPrometheusConditions
+		return ac
+	}
+
+	if prometheusAvailable.Status != monv1.PrometheusConditionTrue {
+		ac.Status = v1alpha1.ConditionStatus(prometheusAvailable.Status)
```

Comment:
though there's a 1:1 mapping for now, it's probably safer to have an explicit map between the 2 types.

Reply:
Its probably best to use `prometheusStatusToMSStatus(prometheusReconciled.Status)` here as well?